Update the docs, change sidebar url property to page/index (#13004)
lampepfl/dotty: docs/docs/usage/scaladoc/settings.md

A directory containing static files from which to generate documentation. Default directory is `./docs`

##### -no-link-warnings

> Avoid warnings for ambiguous and incorrect links in members look up. Doesn't affect warnings for incorrect links of assets etc.

Reviewer: ```suggestion
Suppress warnings for ambiguous or incorrect links in members’ lookup. Doesn't affect warnings for incorrect links of assets etc.
```

Author: Changed

}
```

##### -snippet-compiler

Snippet compiler arguments provide a way to configure snippet checking.

This setting accept list of arguments in format:
args := arg{,arg}
arg := [path=]flag
where path is a prefix of source paths to members to which argument should be set.

If path is not present, argument will be used as default.

Available flags:
compile - Enables snippet checking.
nocompile - Disables snippet checking.

> fail - Enables snippet checking, asserts that snippet doesn't compile.

Reviewer: Could you please run this text through a spell/grammar checker?

Also, I suggest providing a concrete example of usage of this setting.

Author: Added

> You can read more about snippet compiler [here](snippetCompiler.md)

Reviewer: This file is missing

Author: Removed

}
```

##### -snippet-compiler

> Snippet compiler arguments provide a way to configure snippet checking.

Reviewer 2: `checking` sounds a vague term. Compilation is maybe more precise.

Reviewer: “type checking” would also work, imho.

This setting accepts a list of arguments in the format:
args := arg{,arg}
arg := [path=]flag

> where the path is a prefix of source paths to members to which argument should be set.

Reviewer 2: I do not understand this sentence

Author: maybe something like `where 'path' is a prefix of the path to source files where snippets are located and 'flag' is the mode in which snippets will be type checked` would work?

If the path is not present, the argument will be used as the default for all unmatched paths.

Available flags:
compile - Enables snippet checking.
nocompile - Disables snippet checking.
fail - Enables snippet checking, asserts that snippet doesn't compile.

The fail flag comes in handy for snippets that present that some action would eventually fail during compilation, e. g. [Opaques page](docs/reference/other-new-features/opaques.html)

Example usage:

`-snippet-compiler:my/path/nc=nocompile,my/path/f=fail,compile`

Which means:

all snippets in files under directory `my/path/nc` should not be compiled at all
all snippets in files under directory `my/path/f` should fail during compilation
all other snippets should compile successfully

##### -Ysnippet-compiler-debug

> Setting this option causes snippet compiler to print snippet as it is compiled (after wrapping).

Reviewer 2: ```suggestion
Setting this option makes snippet compiler print the snippet as it is compiled (after wrapping).
```

Reviewer 2: There is no documentation about snippet wrapping

##### -Ydocument-synthetic-types

> Documents intrinsic types e. g. Any, Nothing. Setting is useful only for stdlib

Reviewer 2: What does *to document* mean here? Does it annotate them?

Author: Will be `Attach pages with documentation of the intrinsic types e. g. Any, Nothing to the docs. Setting is useful only for stdlib.` ok? The setting is supposed to add few pages to the `API` tree in resulting scaladoc. This is done this way, because there are no tasty files we could create for these types to pass to scaladoc, but for all other users this settings should not be enabled.

Reviewer 2: I would prefer
`Include pages providing documentation for the intrinsic types (e. g. `Any`, `Nothing`) to the docs. Setting is useful only for stdlib.

Reviewer 2: Thank you for explaining. I think you can include what you mentioned in this reply into the doc

Author: Final version:
`Include pages providing documentation for the intrinsic types (e. g. Any, Nothing) to the docs. The setting is useful only for stdlib.
Scaladoc for Scala 3 works for tasty files, but we cannot provide them for intrinsic types since they are embedded in the compiler.
All other users should not concern with this setting.` 

What do you think?

Reviewer 2: > Include pages providing documentation for the intrinsic types (e. g. Any, Nothing) to the docs. The setting is useful only for stdlib because scaladoc for Scala 3 relies on TASTy files, but we cannot provide them for intrinsic types since they are embedded in the compiler. 
All other users should not concern with this setting.

Author: Changed, thanks :D
